src/storage/txn/commands.rs

```diff
@@ -342,19 +342,19 @@ macro_rules! gen_lock {
     };
     ($field: ident) => {
         fn gen_lock(&self, latches: &Latches) -> latch::Lock {
-            latches.gen_lock(&[&self.$field])
+            latches.gen_lock(iter::once(&self.$field))
```

Comment:
@BusyJay
I think (but not sure) `iter::once` might use less memory here?

Reply:
Why? They should be the same.

Reply:
I think it doesn't really matter here

Reply:
Yeah, anything here is just going to get optimized away so it doesn't matter.